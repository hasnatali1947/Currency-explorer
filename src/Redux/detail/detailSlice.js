import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

export const getweatherData = createAsyncThunk(
  'city/weatherData',
  async ({ lat, lon }) => {
    const response = await axios.get(`https://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lon}&appid=bade5986f338cd84a201f41eecd691eb`);
    return response.data;
  },
);

export const fetchCityList = createAsyncThunk(
  'city/fetchCityList',
  async () => {
    const response = await axios.get('https://api.openweathermap.org/data/2.5/find?mode=json&type=like&sort=population&cnt=30&appid=bade5986f338cd84a201f41eecd691eb');
    return response.data.list.map((city) => city.name);
  },
);

const detailSlice = createSlice({
  name: 'detail',
  initialState: {
    cityList: [], // Initialize the cityList property as an empty array
  },
  reducers: {},
  extraReducers(builder) {
    builder
      .addCase(getweatherData.fulfilled, (state, action) => action.payload)
      .addCase(fetchCityList.fulfilled, (state, action) => {
        state.cityList = action.payload;
      });
  },
  reducer(state, action) {
    switch (action.type) {
      default:
        return state;
    }
  },
});

export default detailSlice;
