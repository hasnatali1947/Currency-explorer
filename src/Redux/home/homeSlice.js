/* eslint-disable*/

import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
import axios from "axios";


export const getLatLon = createAsyncThunk(
    'city/lotlon',
    async (cityName) => {
      const response = await axios.get(`https://api.openweathermap.org/data/2.5/weather?q=${cityName}&appid=bade5986f338cd84a201f41eecd691eb`);
      if (response.data === undefined) {
        throw new Error("Invalid city name");
      }
      return response.data;
    }
  );
  


const homeSlice = createSlice ({
        name : 'home',
        initialState : {},
        reducers : {},
        extraReducers(builder){  
            builder.addCase(getLatLon.fulfilled, (state ,action) => {
                return action.payload;
              });
        }
    })

export default homeSlice
