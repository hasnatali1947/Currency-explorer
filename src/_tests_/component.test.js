import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { configureStore } from '@reduxjs/toolkit';
import Home from '../components/home';
import Detail from '../components/detail';
import homeSlice from '../Redux/home/homeSlice';
import detailSlice from '../Redux/detail/detailSlice';
import DisplayWeatheronhome from '../components/displayWeatheronhome';

const store = configureStore({
  reducer: {
    detail: detailSlice.reducer,
    home: homeSlice.reducer,
  },
});

it('test home section', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('tests detail section', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Detail />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('tests detail section', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <DisplayWeatheronhome />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
