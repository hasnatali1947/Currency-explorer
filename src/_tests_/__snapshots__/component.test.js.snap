// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test home section 1`] = `
Array [
  <div
    className="form-container"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        className="searchcity"
        onChange={[Function]}
        placeholder="Search your city"
        value=""
      />
      <button
        type="submit"
      >
        <img
          alt="searchIcon"
          className="searchIcon"
          src="search-icon.png"
          type="submit"
        />
      </button>
    </form>
  </div>,
  <div
    className="topWeatherDisplay"
  >
    <div
      className="topsecond"
    >
      <img
        alt="cloudImg"
        className="TopcloudImg"
        src="weatherIcon.png"
      />
      <div
        className="Weatherdetails"
      >
        <span
          className="countryName"
        >
          Country: PK
        </span>
        <span>
          City: Nowshera
        </span>
        <span>
          Latitude: 33.0281858
        </span>
        <span>
          Longitude: 72.0575345
        </span>
        <span>
          Timezone: 18000
        </span>
      </div>
    </div>
    <a
      href="/detail"
      onClick={[Function]}
    >
      <img
        alt="nextArrow"
        className="nextArrow"
        src="next.png"
      />
    </a>
  </div>,
  <div
    className="homepageWeathers"
  >
    <div
      className="WeatherDisplay first"
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.11)",
        }
      }
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        style={
          Object {
            "cursor": "pointer",
          }
        }
        tabIndex={0}
      >
        <img
          alt="cloudImg"
          className="cloudImg"
          src="weatherIcon.png"
        />
        <div
          className="Weatherdetails"
        >
          <span
            className="countryName"
          >
            Country: FR
          </span>
          <span>
            City: Paris
          </span>
          <span>
            Timezone: 3600
          </span>
        </div>
      </div>
      <a
        href="/detail"
        onClick={[Function]}
      >
        <img
          alt="nextArrow"
          className="nextArrow"
          src="next.png"
        />
      </a>
    </div>
    <div
      className="WeatherDisplay second"
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.065)",
        }
      }
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        style={
          Object {
            "cursor": "pointer",
          }
        }
        tabIndex={0}
      >
        <img
          alt="cloudImg"
          className="cloudImg"
          src="weatherIcon.png"
        />
        <div
          className="Weatherdetails"
        >
          <span
            className="countryName"
          >
            Country: RU
          </span>
          <span>
            City: Moscow
          </span>
          <span>
            Timezone: 10800
          </span>
        </div>
      </div>
      <a
        href="/detail"
        onClick={[Function]}
      >
        <img
          alt="nextArrow"
          className="nextArrow"
          src="next.png"
        />
      </a>
    </div>
    <div
      className="WeatherDisplay third"
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.065)",
        }
      }
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        style={
          Object {
            "cursor": "pointer",
          }
        }
        tabIndex={0}
      >
        <img
          alt="cloudImg"
          className="cloudImg"
          src="weatherIcon.png"
        />
        <div
          className="Weatherdetails"
        >
          <span
            className="countryName"
          >
            Country: AE
          </span>
          <span>
            City: Dubai
          </span>
          <span>
            Timezone: 14400
          </span>
        </div>
      </div>
      <a
        href="/detail"
        onClick={[Function]}
      >
        <img
          alt="nextArrow"
          className="nextArrow"
          src="next.png"
        />
      </a>
    </div>
    <div
      className="WeatherDisplay forth"
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.11)",
        }
      }
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        style={
          Object {
            "cursor": "pointer",
          }
        }
        tabIndex={0}
      >
        <img
          alt="cloudImg"
          className="cloudImg"
          src="weatherIcon.png"
        />
        <div
          className="Weatherdetails"
        >
          <span
            className="countryName"
          >
            Country: US
          </span>
          <span>
            City: New York
          </span>
          <span>
            Timezone: 14400
          </span>
        </div>
      </div>
      <a
        href="/detail"
        onClick={[Function]}
      >
        <img
          alt="nextArrow"
          className="nextArrow"
          src="next.png"
        />
      </a>
    </div>
    <div
      className="WeatherDisplay fifth"
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.11)",
        }
      }
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        style={
          Object {
            "cursor": "pointer",
          }
        }
        tabIndex={0}
      >
        <img
          alt="cloudImg"
          className="cloudImg"
          src="weatherIcon.png"
        />
        <div
          className="Weatherdetails"
        >
          <span
            className="countryName"
          >
            Country: UK
          </span>
          <span>
            City: London
          </span>
          <span>
            Timezone: 28800
          </span>
        </div>
      </div>
      <a
        href="/detail"
        onClick={[Function]}
      >
        <img
          alt="nextArrow"
          className="nextArrow"
          src="next.png"
        />
      </a>
    </div>
    <div
      className="WeatherDisplay sixth"
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.065)",
        }
      }
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        style={
          Object {
            "cursor": "pointer",
          }
        }
        tabIndex={0}
      >
        <img
          alt="cloudImg"
          className="cloudImg"
          src="weatherIcon.png"
        />
        <div
          className="Weatherdetails"
        >
          <span
            className="countryName"
          >
            Country: TR
          </span>
          <span>
            City: Istanbul
          </span>
          <span>
            Timezone: 10800
          </span>
        </div>
      </div>
      <a
        href="/detail"
        onClick={[Function]}
      >
        <img
          alt="nextArrow"
          className="nextArrow"
          src="next.png"
        />
      </a>
    </div>
  </div>,
]
`;

exports[`tests detail section 1`] = `
<div>
  <div
    className="topWeatherInDetail"
  >
    <img
      alt="cloudImg"
      className="cloudImg"
      src="weatherIcon.png"
    />
    <div>
      <span>
        Temperature:
      </span>
      <span>
        All clouds:
      </span>
      <span>
        Wind speed:
      </span>
    </div>
  </div>
  <section
    className="detailSection"
  >
    <p
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.11)",
        }
      }
    >
      Country:
       
    </p>
    <p
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.11)",
        }
      }
    >
      Base:
       
    </p>
    <p
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.11)",
        }
      }
    >
      Longitude:
       
    </p>
    <p
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.11)",
        }
      }
    >
      Latitude:
       
    </p>
    <p
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.11)",
        }
      }
    >
      Temp_min:
       
    </p>
    <p
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.11)",
        }
      }
    >
      Temp_max:
       
    </p>
    <p
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.11)",
        }
      }
    >
      Sunrise:
       
    </p>
    <p
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.11)",
        }
      }
    >
      Sunset:
       
    </p>
    <p
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.11)",
        }
      }
    >
      wind_deg:
       
    </p>
    <p
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.11)",
        }
      }
    >
      wind_gust:
       
    </p>
    <p
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.11)",
        }
      }
    >
      Visibility:
       
    </p>
    <p
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.11)",
        }
      }
    >
      Timezone:
       
    </p>
  </section>
</div>
`;

exports[`tests detail section 2`] = `
<div
  className="homepageWeathers"
>
  <div
    className="WeatherDisplay first"
    style={
      Object {
        "background": "rgba(0, 0, 0, 0.11)",
      }
    }
  >
    <div
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "cursor": "pointer",
        }
      }
      tabIndex={0}
    >
      <img
        alt="cloudImg"
        className="cloudImg"
        src="weatherIcon.png"
      />
      <div
        className="Weatherdetails"
      >
        <span
          className="countryName"
        >
          Country: FR
        </span>
        <span>
          City: Paris
        </span>
        <span>
          Timezone: 3600
        </span>
      </div>
    </div>
    <a
      href="/detail"
      onClick={[Function]}
    >
      <img
        alt="nextArrow"
        className="nextArrow"
        src="next.png"
      />
    </a>
  </div>
  <div
    className="WeatherDisplay second"
    style={
      Object {
        "background": "rgba(0, 0, 0, 0.065)",
      }
    }
  >
    <div
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "cursor": "pointer",
        }
      }
      tabIndex={0}
    >
      <img
        alt="cloudImg"
        className="cloudImg"
        src="weatherIcon.png"
      />
      <div
        className="Weatherdetails"
      >
        <span
          className="countryName"
        >
          Country: RU
        </span>
        <span>
          City: Moscow
        </span>
        <span>
          Timezone: 10800
        </span>
      </div>
    </div>
    <a
      href="/detail"
      onClick={[Function]}
    >
      <img
        alt="nextArrow"
        className="nextArrow"
        src="next.png"
      />
    </a>
  </div>
  <div
    className="WeatherDisplay third"
    style={
      Object {
        "background": "rgba(0, 0, 0, 0.065)",
      }
    }
  >
    <div
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "cursor": "pointer",
        }
      }
      tabIndex={0}
    >
      <img
        alt="cloudImg"
        className="cloudImg"
        src="weatherIcon.png"
      />
      <div
        className="Weatherdetails"
      >
        <span
          className="countryName"
        >
          Country: AE
        </span>
        <span>
          City: Dubai
        </span>
        <span>
          Timezone: 14400
        </span>
      </div>
    </div>
    <a
      href="/detail"
      onClick={[Function]}
    >
      <img
        alt="nextArrow"
        className="nextArrow"
        src="next.png"
      />
    </a>
  </div>
  <div
    className="WeatherDisplay forth"
    style={
      Object {
        "background": "rgba(0, 0, 0, 0.11)",
      }
    }
  >
    <div
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "cursor": "pointer",
        }
      }
      tabIndex={0}
    >
      <img
        alt="cloudImg"
        className="cloudImg"
        src="weatherIcon.png"
      />
      <div
        className="Weatherdetails"
      >
        <span
          className="countryName"
        >
          Country: US
        </span>
        <span>
          City: New York
        </span>
        <span>
          Timezone: 14400
        </span>
      </div>
    </div>
    <a
      href="/detail"
      onClick={[Function]}
    >
      <img
        alt="nextArrow"
        className="nextArrow"
        src="next.png"
      />
    </a>
  </div>
  <div
    className="WeatherDisplay fifth"
    style={
      Object {
        "background": "rgba(0, 0, 0, 0.11)",
      }
    }
  >
    <div
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "cursor": "pointer",
        }
      }
      tabIndex={0}
    >
      <img
        alt="cloudImg"
        className="cloudImg"
        src="weatherIcon.png"
      />
      <div
        className="Weatherdetails"
      >
        <span
          className="countryName"
        >
          Country: UK
        </span>
        <span>
          City: London
        </span>
        <span>
          Timezone: 28800
        </span>
      </div>
    </div>
    <a
      href="/detail"
      onClick={[Function]}
    >
      <img
        alt="nextArrow"
        className="nextArrow"
        src="next.png"
      />
    </a>
  </div>
  <div
    className="WeatherDisplay sixth"
    style={
      Object {
        "background": "rgba(0, 0, 0, 0.065)",
      }
    }
  >
    <div
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "cursor": "pointer",
        }
      }
      tabIndex={0}
    >
      <img
        alt="cloudImg"
        className="cloudImg"
        src="weatherIcon.png"
      />
      <div
        className="Weatherdetails"
      >
        <span
          className="countryName"
        >
          Country: TR
        </span>
        <span>
          City: Istanbul
        </span>
        <span>
          Timezone: 10800
        </span>
      </div>
    </div>
    <a
      href="/detail"
      onClick={[Function]}
    >
      <img
        alt="nextArrow"
        className="nextArrow"
        src="next.png"
      />
    </a>
  </div>
</div>
`;
